File: src/main/webapp/js/Dot.js
class Dot {
    static type = "Dot"

    constructor (options) {
        this.name =     options.name
        this.x =        options.x
        this.y =        options.y
    }

    sout() {
        console.log("Name: " + this.name + " x: " + this.x + " y: " + this.y)
    }
}

File: src/main/java/com/example/jspweb/Logic/Dot.java
package com.example.jspweb.Logic;

public class Dot {
    String name;
    String x;
    String y;

    public Dot() {
    }

    public double getX() {return Double.parseDouble(this.x);}
    public double getY() {return Double.parseDouble(this.y);}

    @Override
    public String toString() {
        return "Dot{" +
                "name='" + name + '\'' +
                ", x='" + x + '\'' +
                ", y='" + y + '\'' +
                '}';
    }
}
